packages/common-all/src/utils.ts

```diff
@@ -1267,6 +1268,37 @@ export class ConfigUtils {
       return { isValid, minCompatClientVersion, minCompatConfigVersion };
     }
   }
+
+  static detectMissingDefaults(opts: {
+    config: Partial<IntermediateDendronConfig>;
+    defaultConfig?: IntermediateDendronConfig;
+  }): RespV3<{
+    needsBackfill: boolean;
+    backfilledConfig: IntermediateDendronConfig;
+  }> {
+    try {
+      const { config } = opts;
+      const configDeepCopy = _.cloneDeep(config);
+      let { defaultConfig } = opts;
+      if (defaultConfig === undefined) {
+        defaultConfig = ConfigUtils.genDefaultConfig();
+      }
+      const backfilledConfig = _.defaultsDeep(config, defaultConfig);
+      return {
+        data: {
+          needsBackfill: !_.isEqual(backfilledConfig, configDeepCopy),
+          backfilledConfig,
+        },
+      };
+    } catch (error) {
+      return {
```

Comment:
are we expecting an error here?

Reply:
in that case, just return the value and not use `RespV3`. `RespV3` is when we expect errors or are making a network call

Reply:
okay, I'll fix that